vllm/utils/__init__.py

```diff
@@ -2732,6 +2738,15 @@ def measure(self):
             "allocated_bytes.all.peak", 0)
 
         self.free_memory, self.total_memory = torch.cuda.mem_get_info()
+        shared_sysmem_device_mem_sms = (87, 110, 121)  # Orin, Thor, Spark
+        if self._get_device_sm() in shared_sysmem_device_mem_sms:
+            # On these devices, which use sysmem as device mem,
+            # torch.cuda.mem_get_info() only reports "free" memory,
+            # which can be lower than what is actually
+            # available due to not including cache memory.
+            # So we use the system available memory metric instead.
+            self.free_memory = psutil.virtual_memory().available
```

Comment:
How is virtual memory helpful here? Isn;t that referring to CPU virtual memory - what does that have to do with free device (GPU) memory?

Reply:
@johnnynunez

Reply:
You mean a screenshot? I can give you one

Reply:
Yep that sounds good, can you add that as a comment in the code?

Reply:
Give me minute